tests/test_perf_analyzer_config_generator.py

```diff
@@ -0,0 +1,360 @@
+        self._run_and_test_perf_analyzer_config_generator(
+            yaml_content, expected_configs, pa_cli_args)
+
+    def test_batch_size_search_enabled(self):
+        ''' 
+        Test: 
+            - Schmoo batch sizes
+            - Run Config Search enabled
+        
+        Batch sizes: 1,2,4
+        Concurrency: log2(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY)+1
+        '''
+
+        # yapf: disable
+        yaml_content = convert_to_bytes("""
+            profile_models:
+                - my-model
+            """)
+        # yapf: enable
+
+        batch_sizes = [1, 2, 4]
+        concurrencies = utils.generate_log_list(
+            DEFAULT_RUN_CONFIG_MAX_CONCURRENCY)
+        expected_configs = [
+            self._create_expected_config(batch_size=b, concurrency=c)
+            for b in batch_sizes
+            for c in concurrencies
+        ]
+
+        pa_cli_args = ['-b 1,2,4']
+        self._run_and_test_perf_analyzer_config_generator(
+            yaml_content, expected_configs, pa_cli_args)
+
+    def test_concurrency(self):
+        ''' 
+        Test: 
+            - Schmoo concurrency
+            - Test with auto-search enabled & disabled
+        
+        Concurrency: 1-4
+        Default (1) value will be used for batch size 
+        and 4 configs will be generated 
+        '''
+
+        # yapf: disable
+        yaml_content = convert_to_bytes("""
+            profile_models:
+                - my-model
+            """)
+        # yapf: enable
+
+        concurrencies = [1, 2, 3, 4]
+        expected_configs = [
+            self._create_expected_config(concurrency=c) for c in concurrencies
+        ]
+
+        pa_cli_args = ['-c 1,2,3,4']
+        self._run_and_test_perf_analyzer_config_generator(
+            yaml_content, expected_configs, pa_cli_args)
+
+        pa_cli_args = ['-c 1,2,3,4', '--run-config-search-disable']
+        self._run_and_test_perf_analyzer_config_generator(
+            yaml_content, expected_configs, pa_cli_args)
+
+    def test_batch_size_and_concurrency(self):
+        '''
+        Test:
+            - Schmoo batch sizes and concurrency
+            - Run Config Search enabled & disabled
+
+        Batch sizes: 1,2,4
+        Concurrency: 1-4
+        
+
+        12 configs will be generated
+        '''
+
+        # yapf: disable
+        yaml_content = convert_to_bytes("""
+            profile_models:
+                - my-model
+            """)
+        # yapf: enable
+
+        batch_sizes = [1, 2, 4]
+        concurrencies = [1, 2, 3, 4]
+
+        expected_configs = [
+            self._create_expected_config(batch_size=b, concurrency=c)
+            for b in batch_sizes
+            for c in concurrencies
+        ]
+
+        pa_cli_args = ['-b 1,2,4', '-c 1,2,3,4']
+        self._run_and_test_perf_analyzer_config_generator(
+            yaml_content, expected_configs, pa_cli_args)
+
+        pa_cli_args = ['-b 1,2,4', '-c 1,2,3,4', '--run-config-search-disable']
```

Comment:
I see tests with `--run-config-search-disable`. I don't see any that set `--run-config-search-max-concurrency`

Reply:
See my previous comment.